app/services/create_user_service.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+class CreateUserService
+  def self.call(**args)
+    new(**args).call
+  end
+
+  attr_reader :user, :params, :password, :consent, :language_ids, :skill_ids,
+              :interest_ids, :image_tokens
+
+  def initialize(params:, password:, consent:, language_ids:, skill_ids:, interest_ids:, image_tokens:) # rubocop:disable Metrics/LineLength
+    @params = params
+    @password = password
+    @consent = consent
+    @language_ids = language_ids
+    @skill_ids = skill_ids
+    @interest_ids = interest_ids
+    @image_tokens = image_tokens
+    @user = User.new(params)
+  end
+
+  def call
+    user.password = password
+
+    user.validate
+
+    unless accepted_terms?
+      consent_error = I18n.t('errors.user.must_consent_to_terms_of_agreement')
+      user.errors.add(:consent, consent_error)
+      return user
+    end
+
+    return user unless user.valid?
+
+    begin
+      commit!
+      user
+    rescue ActiveRecord::RecordInvalid,
+           ActiveRecord::Rollback,
+           ActiveRecord::StatementInvalid,
+           ActiveRecord::RecordNotSaved
```

Comment:
Consider manually invoking `ErrorNotifier`, so that we can track how often this happens.